src/analysis/calculus/mean_value.lean

```diff
@@ -573,7 +573,48 @@ begin
   exact ⟨c, cmem, sub_eq_zero.1 hc⟩
 end
 
-omit hgc hgg'
+omit hfc hgc
+
+/-- Cauchy's Mean Value Theorem, extended `has_deriv_at` version. -/
+lemma exists_ratio_has_deriv_at_eq_ratio_slope' {lfa lga lfb lgb : ℝ}
+  (hff' : ∀ x ∈ Ioo a b, has_deriv_at f (f' x) x) (hgg' : ∀ x ∈ Ioo a b, has_deriv_at g (g' x) x)
+  (hfa : tendsto f (nhds_within a $ Ioi a) (nhds lfa)) (hga : tendsto g (nhds_within a $ Ioi a) (nhds lga))
+  (hfb : tendsto f (nhds_within b $ Iio b) (nhds lfb)) (hgb : tendsto g (nhds_within b $ Iio b) (nhds lgb)) :
+  ∃ c ∈ Ioo a b, (lgb - lga) * (f' c) = (lfb - lfa) * (g' c) :=
+begin
+  let h := λ x, (lgb - lga) * f x - (lfb - lfa) * g x,
+  have hha : tendsto h (nhds_within a $ Ioi a) (nhds $ lgb * lfa - lfb * lga),
```

Comment:
We really need a tactic doing this kind of computation, this is ridiculous...

Reply:
That would be great indeed